### PR TITLE
feat: LakeTagInput

### PR DESCRIPTION
**Pull request by ncomont**

### CODE REVIEW

File: packages/lake/src/components/LakeTagInput.tsx [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+      onHoverEnd: () => setIsHovered(false),
+    });
+
+    const onTextInputChange = useCallback((value: string) => {
+      const values = [...new Set(value.split(SEPARATORS_REGEX).filter(s => s.length))];
+      if (values.length > 1 || values[0] !== value) {
+        onAddValues(values);
+        inputRef.current?.clear();
+      }
+    }, []);
```
Comment (bloodyowl) [inline packages/lake/src/components/LakeTagInput.tsx:139]: small warning here

File: packages/lake/src/components/LakeTagInput.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+          <TextInput
+            ref={mergedRef}
+            style={[styles.input, disabled && styles.disabled]}
+            onFocus={onFocus}
+            onBlur={onBlur}
+            aria-disabled={disabled}
+            onChangeText={onTextInputChange}
+            readOnly={readOnly}
+          />
```
Comment (bloodyowl) [inline packages/lake/src/components/LakeTagInput.tsx:200]: maybe add support for a placeholder here

File: packages/lake/src/components/LakeTagInput.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+          {values.map((v, i) => (
+            <Tag
+              key={i}
+              onPressRemove={!readOnly && !disabled ? () => onRemoveValue(v) : undefined}
+              style={styles.tag}
+              color={validator(v) ? "gray" : "negative"}
+            >
+              {v}
+            </Tag>
+          ))}
```
Comment (bloodyowl) [inline packages/lake/src/components/LakeTagInput.tsx:190]: Screenshot 2023-12-15 at 14 47 01

tags can overflow, shouldn't we give them a maxWidth?

**Review by bloodyowl — COMMENTED** [review]
looks good!

just a few comments, and UX-wise, could we add an handler to make `Enter` keystroke add the value? Add maybe `Backspace` remove the last tag before the input?

**Review by sandrine-ds — APPROVED** [review]

**Pull request merged**